packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -113,6 +117,18 @@ class HotRunner extends ResidentRunner {
     for (FlutterDevice device in flutterDevices)
       device.initLogReader();
 
+    if (previewDart2 && (generator == null)) {
+      generator = new ResidentCompiler(
+        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath));
+      final String outputFile = await generator.compile(mainPath);
+      // If compilation was successful, no reason VM should reject it.
```

Comment:
This is supposed to be first compilation of given script and VM has nothing loaded yet. So I'm assuming script will be accepted.